submodules/power_sap_instance_init/scripts/init_powervs.sh

```diff
@@ -179,20 +179,89 @@ fi
 # RHEL Setup                              #
 ###########################################
 if [ "$OS_DETECTED" == "RHEL" ]; then
-
+   FILE="/etc/bashrc"
+   if [[ -n $no_proxy_ip ]] ; then
+        grep -qx "export no_proxy=$no_proxy_ip" "$FILE"                 || echo "export no_proxy=$no_proxy_ip"              >> "$FILE"
+        source /etc/bashrc
+   fi
    if [[ $proxy_ip_and_port =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+:[0-9]+$ ]] ; then
 
      echo "Setting exports in /etc/bashrc and /etc/dnf file On RHEL"
-     FILE="/etc/bashrc"
      grep -qx "export http_proxy=http://$proxy_ip_and_port" "$FILE"  || echo "export http_proxy=http://$proxy_ip_and_port"  >> "$FILE"
      grep -qx "export https_proxy=http://$proxy_ip_and_port" "$FILE" || echo "export https_proxy=http://$proxy_ip_and_port" >> "$FILE"
      grep -qx "export HTTP_PROXY=http://$proxy_ip_and_port" "$FILE"  || echo "export HTTP_PROXY=http://$proxy_ip_and_port"  >> "$FILE"
      grep -qx "export HTTPS_PROXY=http://$proxy_ip_and_port" "$FILE" || echo "export HTTPS_PROXY=http://$proxy_ip_and_port" >> "$FILE"
-     grep -qx "export no_proxy=$no_proxy_ip" "$FILE"                 || echo "export no_proxy=$no_proxy_ip"                 >> "$FILE"
      grep -qx "proxy=http://$proxy_ip_and_port"  /etc/dnf/dnf.conf   || echo "proxy=http://$proxy_ip_and_port"              >> /etc/dnf/dnf.conf
-      ###### Restart Network #######
-
-      /usr/bin/systemctl restart NetworkManager
+     source /etc/bashrc
+     OS_Activated="$(subscription-manager status | grep -ic "Overall Status: Current")"
+     if [ "$OS_Activated" -ge 1 ] ; then
+        echo "OS is Registered"
+     else
+        ARCH=$(uname -p)
+      ##### check if the system is a x86_64 processor VM
+        if [[ "$ARCH" == "x86_64" ]]; then
+      #### Wait for registration to complete
+ 	     count=0
+             while [[ $count -le 15 ]]; do
+                sleep 60
+                OS_Activated="$(subscription-manager status | grep -c "Overall Status: Current")"
+                if  [[ "$OS_Activated" -ge 1 ]] ; then
+        	        echo "OS is Registered"
+    		          break;
+	              fi
```

Comment:
can you fix the indentation here so it is consistent across the script?

Reply:
Done. Added shell extension. Thank you :)